samza-core/src/main/java/org/apache/samza/startpoint/StartpointManager.java

```diff
@@ -43,64 +45,115 @@
   private final StartpointSerde startpointSerde = new StartpointSerde();
 
-  private boolean stopped = false;
+  private boolean stopped = true;
 
   /**
    *  Constructs a {@link StartpointManager} instance by instantiating a new metadata store connection.
-   *  This is primarily used for testing.
+   *  This should only be used for testing.
    */
   @VisibleForTesting
   StartpointManager(MetadataStoreFactory metadataStoreFactory, Config config, MetricsRegistry metricsRegistry) {
     Preconditions.checkNotNull(metadataStoreFactory, "MetadataStoreFactory cannot be null");
     Preconditions.checkNotNull(config, "Config cannot be null");
     Preconditions.checkNotNull(metricsRegistry, "MetricsRegistry cannot be null");
 
+    this.manageMetadataStoreLifecyle = true;
     this.metadataStore = metadataStoreFactory.getMetadataStore(NAMESPACE, config, metricsRegistry);
-    LOG.info("StartpointManager created with metadata store: {}", metadataStore.getClass().getCanonicalName());
-    this.metadataStore.init();
+    this.readWriteStore = new NamespaceAwareCoordinatorStreamStore(metadataStore, NAMESPACE);
+    this.fanOutStore = new NamespaceAwareCoordinatorStreamStore(metadataStore, NAMESPACE_FAN_OUT);
+    LOG.info("Managing lifecycle of metadata store: {}", metadataStore.getClass().getCanonicalName());
+    LOG.info("Startpoints are written to namespace: {} and fanned out to namespace: {} in the metadata store", NAMESPACE,
+        NAMESPACE_FAN_OUT);
   }
 
   /**
    *  Builds the StartpointManager based upon the provided {@link MetadataStore} that is instantiated.
    *  Setting up a metadata store instance is expensive which requires opening multiple connections
-   *  and reading tons of information. Fully instantiated metadata store is taken as a constructor argument
+   *  and reading tons of information. Fully instantiated metadata store is passed in as a constructor argument
    *  to reuse it across different utility classes.
    *
    * @param metadataStore an instance of {@link MetadataStore} used to read/write the start-points.
    */
   public StartpointManager(MetadataStore metadataStore) {
     Preconditions.checkNotNull(metadataStore, "MetadataStore cannot be null");
-    this.metadataStore = new NamespaceAwareCoordinatorStreamStore(metadataStore, NAMESPACE);
+
+    this.manageMetadataStoreLifecyle = false;
+    this.metadataStore = metadataStore;
+    this.readWriteStore = new NamespaceAwareCoordinatorStreamStore(metadataStore, NAMESPACE);
+    this.fanOutStore = new NamespaceAwareCoordinatorStreamStore(metadataStore, NAMESPACE_FAN_OUT);
+    LOG.info("Lifecycle of metadata store: {} is managed externally", metadataStore.getClass().getCanonicalName());
+    LOG.info("Startpoints are written to namespace: {} and fanned out to namespace: {} in the metadata store", NAMESPACE,
+        NAMESPACE_FAN_OUT);
   }
 
+  /**
+   * Perform startup operations.
+   */
   public void start() {
```

Comment:
This is related to a reply on a different comment, but putting directly here too. I think you will need to manage the lifecycle of `fanOutStore` and `readWriteStore`, since you created those within this object.